src/gui/plugins/scene3d/Scene3D.cc

```diff
@@ -2730,20 +2730,23 @@ void Scene3D::Update(const UpdateInfo &_info,
           return true;
         });
 
-    renderWindow->SetWorldName(this->dataPtr->worldName);
-    auto worldEntity =
-      _ecm.EntityByComponents(components::Name(this->dataPtr->worldName),
-        components::World());
-    auto renderEngineGuiComp =
-      _ecm.Component<components::RenderEngineGuiPlugin>(worldEntity);
-    if (renderEngineGuiComp && !renderEngineGuiComp->Data().empty())
+    if (!this->dataPtr->worldName.empty())
     {
-      this->dataPtr->renderUtil->SetEngineName(renderEngineGuiComp->Data());
-    }
-    else
-    {
-      igndbg << "RenderEngineGuiPlugin component not found, "
-        "render engine won't be set from the ECM" << std::endl;
+      renderWindow->SetWorldName(this->dataPtr->worldName);
+      auto worldEntity =
+        _ecm.EntityByComponents(components::Name(this->dataPtr->worldName),
+          components::World());
```

Comment:
This is not introduced by this PR, but the `worldEntity` could be got from the `Each` call above on line 2724.

Reply:
updated. iche033/scene3d_world_name